src/libraries/Microsoft.PowerFx.Core/Public/Types/OptionSetValueType.cs

```diff
@@ -1,12 +1,19 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
 using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Public.Values;
 using Microsoft.PowerFx.Core.Types;
 using Microsoft.PowerFx.Core.Utils;
 
 namespace Microsoft.PowerFx.Core.Public.Types
 {
+    /// <summary>
+    /// Power Fx type for an enum-like things such as OptionSets. 
+    /// </summary>
     public class OptionSetValueType : FormulaType
```

Comment:
>OptionSetValueType [](http://example.com/codeflow?start=17&length=18)

This (lives in Core) is very similar to OptionSet (lives in interpreter). 


OptionSet is just an impl of IExternalOptionSet which is used to instantiate a OSVT.  OS even gives you a OSVT - but OSVT can't give you back the OS (at best, it just has the IExternalOptionSet interface, which is buried in the DType). 


So it's more valuable to have useful methods (like TryGetValue) here than in OS.  Here, they're available in Core and work with dataverse and other impls.